Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,301 @@
+                }
+
+                var lossForCurrentPerson  =  CalculateLossForPerson(allAmounts);
+
+                if (lossForCurrentPerson > HighestLossEver)
+                {
+                    HighestLossEver = lossForCurrentPerson;
+                    PersonWithHighestLoss = currentPersonData[0];
+                }
+            }
+
+            var answer = $"{PersonWithHighestLoss} lost the most money. -¤{HighestLossEver}.";
+
+            return answer;
+
+        }
+
+        public static decimal CalculateLossForPerson(string [] data) 
+        {
+            var lossForCurrentPerson = 0M;
+
+            for (int j = 0; j < data.Length - 1; j++)
+            {
+                if(!decimal.TryParse(data[j], out decimal amount1))
+                {
+                    break; 
+                }
+
+                if (!decimal.TryParse(data[j + 1], out decimal amount2))
+                {
+                    break;
+                }
+                lossForCurrentPerson = amount1 - amount2;
+            }
+
+            return lossForCurrentPerson; 
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var HighestBalance = decimal.MinValue;
+            var RichestPerson = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+
+                if (!decimal.TryParse(currentPersonData[currentPersonData.Length - 1], out decimal highestAmountOfPerson))
+                {
+                    return "N/A.";
+                }
+
+                if (highestAmountOfPerson == HighestBalance)
+                {
+                    if (i == peopleAndBalances.Length - 1)
+                    {
+                        RichestPerson.Append(" and ");
+                    }
+                    else
+                    {
+                        RichestPerson.Append(", ");
+                    }
+
+                    RichestPerson.Append(currentPersonData[0]);
+                }
+
+                HighestBalance = FindMorePeopleWithSameAmount(HighestBalance, RichestPerson, currentPersonData, highestAmountOfPerson);
+
+            }
+
+            var answer = $"{RichestPerson.ToString()} is the richest person. ¤{HighestBalance}.";
+
+            return answer;
+
+
+        }
+
+        private static decimal FindMorePeopleWithSameAmount(decimal HighestBalance, StringBuilder RichestPerson, string[] currentPersonData, decimal highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson > HighestBalance)
+            {
+                HighestBalance = highestAmountOfPerson;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'HighestBalance'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)